python/tvm/topi/arm_cpu/mprofile/dsp/micro_kernel/multi_channel_convolve.py

```diff
@@ -179,20 +179,26 @@ def _dual_int16_channel_convolve_impl(_tensor_h, tensor_w, channels, kernel_h, k
         extern "C"
         #endif
         int32_t {_get_func_name("int16", tensor_w, channels, kernel_h, kernel_w, suffix)}(
-            uint32_t *out,
-            uint32_t *tensor,
-            uint32_t *kernel) {{
+            int32_t *out,
+            int16_t *tensor,
+            int16_t *kernel) {{
 
-          uint32_t sum_c0 = 0;
-          uint32_t sum_c1 = 0;
+          int32_t sum_c0 = 0;
+          int32_t sum_c1 = 0;
+
+          int32_t kernel_i32[{kernel_h} * {kernel_w}];
+          memcpy(kernel_i32, kernel, {kernel_h} * {kernel_w} * 4);
```

Comment:
nit: can we please add `sizeof()` instead of using 4 directly?

Reply:
Same applies to the code below where 2 and 4 are used.

Reply:
`sizeof(int32_t)` instead of `4` makes sense, but why `2`?

Reply:
I guess the same reason applies. For clarity, we could show the derivation of constants here.

Reply:
my bad - `2` is part of the compute itself as clarified by @NicolaLancellotti offline.